Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1186,9 +1205,51 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     m_backgroundWebView = nil;
 }
 
+void WebExtensionContext::loadBackgroundPageListenersFromStorage()
+{
+    NSData *listenersData = [m_state objectForKey:backgroundPageListenersKey];
+    NSCountedSet *savedListeners = [NSKeyedUnarchiver unarchivedObjectOfClasses:[NSSet setWithArray:@[ NSCountedSet.class, NSNumber.class ]] fromData:listenersData error:nil];
+
+    m_backgroundPageListeners.clear();
+    for (NSNumber *entry in savedListeners)
+        m_backgroundPageListeners.add(static_cast<WebExtensionEventListenerType>(entry.unsignedIntValue), [savedListeners countForObject:entry]);
+}
+
+void WebExtensionContext::saveBackgroundPageListenersToStorage()
+{
+    if (backgroundContentURL().isEmpty() || isPersistent())
+        return;
+
+    if (!m_state.get().count)
+        return;
```

Comment:
This should be removed. It won't save anything if state is empty, and we want to be able to save state the first time.